### PR TITLE
DynamoDB streaming binary content

### PR DESCRIPTION
**Pull request by giograno**
This PR addresses #6364 where the user reported error messages in the logs while trying to stream the DDB changes to Kinesis. This was due to errors in serializing bytes content. 
Introduced a new encoder for binary content (less broad than the `CustomEncoder` we already have) and introduced/refactored some tests.

### CODE REVIEW

**Review by whummer — COMMENTED** [review]
Thanks for tackling this @giograno ! Added a small question/comment regarding binary data with non-printable characters..

File: tests/integration/test_dynamodb.py [hunk 1]
```diff
@@ -499,6 +500,37 @@ def test_batch_write_binary(self, dynamodb_client):
...
+            ProvisionedThroughput={"ReadCapacityUnits": 5, "WriteCapacityUnits": 5},
+            StreamSpecification={
+                "StreamEnabled": True,
+                "StreamViewType": "NEW_AND_OLD_IMAGES",
+            },
+        )
+        stream_name = get_kinesis_stream_name(table_name)
+        wait_for_stream_ready(stream_name)
+        response = dynamodb_client.put_item(
+            TableName=table_name, Item={"id": {"S": "id1"}, "data": {"B": b"binary_data"}}
```
[the remark this answers is not in the document]
Comment (giograno) [reply tests/integration/test_dynamodb.py]: Good point. `to_str()` was indeed failing for non-printable binaries. Added a `errors="replace"` while calling the decode method.

**Comment by thrau** [comment]
also added @alexrashed as reviewer since this relates to ASF

**Review by alexrashed — APPROVED** [review]
LGTM! Also, this shouldn't affect any ASF serializer or parser (since it basically only affects client calls and the parsers and serialziers traverse the hierarchy and pre-process individual fields depending on their type).

**Pull request merged**